src/Adapters/Storage/Postgres/PostgresStorageAdapter.js

```diff
@@ -1775,6 +1789,36 @@ function processRegexPattern(s) {
   return literalizeRegexPart(s);
 }
 
+function isStartsWithRegex(value) {
+  if (!value || typeof value !== 'string' || !value.startsWith('^')) {
+    return false;
+  }
+
+  const matches = value.match(/\^\\Q.*\\E/);
+  return !!matches;
+}
+
+function isAllValuesRegexOrNone(values) {
+  if (values.length == 0) {
+    return true;
```

Comment:
Just looking over the code coverage. I don't see a test case for None / Empty {}